src/utils.c

```diff
@@ -363,3 +363,84 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// if (length(x)) length(x[[1L]]) else 0L
+int NROW(SEXP x) {
+  if (!LENGTH(x))
+    return 0; // # nocov # not yet reached from anywhere, cbindlist uses it but escapes for !NCOL(x)
+  return length(VECTOR_ELT(x, 0));
+}
+
+// length(x)
+int NCOL(SEXP x) {
+  return LENGTH(x);
+}
+
+/*// c("data.table","data.frame")
+static SEXP char2_dtdf() {
+  SEXP char2_dtdf = PROTECT(allocVector(STRSXP, 2));
+  SET_STRING_ELT(char2_dtdf, 0, char_datatable);
+  SET_STRING_ELT(char2_dtdf, 1, char_dataframe);
+  UNPROTECT(1);
+  return char2_dtdf;
+}
+
+// .set_row_names(x)
+static SEXP set_row_names(int n) {
+  SEXP ans = R_NilValue;
+  if (n) {
+    ans = PROTECT(allocVector(INTSXP, 2));
+    INTEGER(ans)[0] = NA_INTEGER;
+    INTEGER(ans)[1] = -n;
+  } else {
+    ans = PROTECT(allocVector(INTSXP, 0));
+  }
+  UNPROTECT(1);
+  return ans;
+}
+
+// setDT(x) ## not in-place!
+SEXP setDT(SEXP x) {
+  if (!isNewList(x))
+    error("internal error: C setDT should be called only on a list"); // # nocov
+  setAttrib(x, R_ClassSymbol, char2_dtdf());
+  setAttrib(x, sym_rownames, set_row_names(NROW(x)));
+  return alloccolwrapper(x, GetOption(sym_alloccol, R_NilValue), GetOption(sym_verbose, R_NilValue));
+}*/
+
+// inherits(x, "data.table")
+bool isDataTable(SEXP x) {
```

Comment:
i could have sworn we have such helper already...

Reply:
I have seen `bool isDataTable = INHERITS()`, so not a function